components/gantry/oneaxis/oneaxis.go

```diff
@@ -167,6 +166,48 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 	return oAx, nil
 }
 
+func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	if g.lengthMm != newConf.LengthMm {
+		g.lengthMm = newConf.LengthMm
+	}
+	if g.rpm != newConf.GantryRPM {
+		g.rpm = newConf.GantryRPM
+	}
+	if g.mmPerRevolution != newConf.MmPerRevolution {
+		g.mmPerRevolution = newConf.MmPerRevolution
+	}
+
+	// Rerun homing if anything with the limit switch pins changes
+	if (len(g.limitSwitchPins) != len(newConf.LimitSwitchPins)) || (g.limitHigh != *newConf.LimitPinEnabled) {
```

Comment:
If the list of limit switches has changed, I think we should store the updated list somewhere, but I don't see where we're doing that already.

A useful pattern Eric turned me on to is that the constructor (`newOneAxis`) shouldn't store any data itself, and instead it should just call `Reconfigure` as one of the last things it does, and then `Reconfigure` is where we actually build out all the state from the config data. That way, we can be confident that `Reconfigure` stores all the things it needs to store, because you can't even build the component in the first place without it.

So, I'd kinda expect a 4-line change somewhere within `newOneAxis` to just call `Reconfigure`, then a `}` to end the function, then the start of the `Reconfigure` function, and the rest of the code that used to be in `newOneAxis`.